hypergpufunctions.cu

```diff
@@ -140,58 +189,77 @@ void HyperFunctionsGPU::spec_sim_GPU() {
 
     cudaMemcpy(out, d_out, sizeof(int) * N_size, cudaMemcpyDeviceToHost); 
 
-    Mat test_img1(mlt1[1].rows, mlt1[1].cols, CV_8UC1, Scalar(0));
-    spec_simil_img=test_img1;
-    this->oneD_array_to_mat(out);
-            
-            
+    //Mat test_img1(mlt1[1].rows, mlt1[1].cols, CV_8UC1, Scalar(0));
+    //spec_simil_img=test_img1;
+    this->oneD_array_to_mat(out, mlt1[1].rows, mlt1[1].cols);   
 }
+
 void HyperFunctionsGPU::deallocate_memory() 
 {
     cudaFree(d_img_array); cudaFree(d_ref_spectrum); cudaFree(d_out);
 }
 
-
-
+/* allocating CUDA memory. 
+* In cuda, grids contain blocks of threads, which are used for parallel computations. 
+* One pixel in img_array needs one thread for a computation, so threads = number of pixels. 
+* Set the number of threads per block to be 512 - this is the maximum threads per block for older GPUs.
+* Grid size is the number of of blocks, given by the number of threads / block size + 1. 
+*/
 void HyperFunctionsGPU::allocate_memory(int* img_array) {
-    //allocating CUDA memory
+    
     N_points=mlt1[1].rows*mlt1[1].cols*mlt1.size(); 
     N_size=mlt1[1].rows*mlt1[1].cols;    
     num_lay=  mlt1.size();
     block_size = 512;
-    grid_size = ((N_points + block_size) / block_size);
-
+    grid_size = ((N_points + block_size) / block_size); 
+    
     int tmp_len1=reference_spectrums[ref_spec_index].size(); 
 
-    cudaHostAlloc ((void**)&out, sizeof(int) *N_size, cudaHostAllocDefault);
+    cudaHostAlloc ((void**)&out, sizeof(unsigned int) * N_size, cudaHostAllocDefault);
```

Comment:
what affect does changing to unsigned int have?

Reply:
woops I was trying to make casting to 8UC1 work, forgot to change this back